### PR TITLE
Remove `http.request.socket` fields

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Remove some unused, non-indexed fields, to tidy up our data model and bring it more in line with ECS.

Stop recording `http.request.socket.encrypted`, with no replacement. This field is not used in the UI, and is redundant; the URL scheme can be used instead.

Stop recording `http.request.socket.remote_address`. Instead, record the direct network peer IP and port in `source.ip` and `source.port`.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~

## How to test these changes

1. Run `apm-server -E instrumentation.enabled=true`
2. `curl -H "X-Forwarded-For: 1.2.3.4" http://localhost:8200/foo`
3. Wait for a transaction to be indexed
4. Check that the transaction does _not_ have any `http.request.socket` fields, but has `source.ip` (127.0.0.1) and `source.port` set (client's bind port, pass `--local-port` to curl to control this) and `client.ip` (1.2.3.4)

## Related issues

https://github.com/elastic/apm-server/issues/4410
https://github.com/elastic/apm-server/issues/4120

### CODE REVIEW

**Comment by axw** [comment]
@simitt @felixbarny I'd like to tidy up out model now, before we commit to something long-term in https://github.com/elastic/apm-server/issues/4120. Please let me know if you have any concerns about these changes:
 - ignoring `context.request.socket.encrypted` as it can be derived from the URL scheme
 - translating `context.request.socket.remote_address` to `source.ip`
 - (for RUM) recording the address of the direct network peer (e.g. reverse proxy) as `source.ip` and `source.port`; X-Forwarded-For etc. are used only for setting `client.ip`

**Review by felixbarny — APPROVED** [review]
Thanks for cleaning up the data model. No objections on removing the socket fields.

**Review by simitt — COMMENTED** [review]
For backend agents, this change mostly means losing the information about the original source right? (I might be missing something).

File: beater/request/context.go [hunk 1]
```diff
@@ -76,7 +87,8 @@ func (c *Context) Reset(w http.ResponseWriter, r *http.Request) {
 	c.w = w
 	c.Request = r
 	if r != nil {
```
[the remark this answers is not in the document]
Comment (axw) [reply beater/request/context.go:91]: No harm, updated

**Comment by axw** [comment]
> For backend agents, this change mostly means losing the information about the original source right? (I might be missing something).

We shouldn't be losing information. `context.http.socket.remote_addr` is now mapped to `source.ip`: https://github.com/axw/apm-server/blob/48b841ebe94a45c75d21f597d686ce7bbd3caf53/model/modeldecoder/v2/decoder.go#L227

**Comment by simitt** [comment]
> We shouldn't be losing information. context.http.socket.remote_addr is now mapped to source.ip: https://github.com/axw/apm-server/blob/48b841ebe94a45c75d21f597d686ce7bbd3caf53/model/modeldecoder/v2/decoder.go#L227

But only if not already set from the request's `RemoteAddr` when processing the incoming request.

**Comment by axw** [comment]
@simitt we only set `source.ip` from the agent's HTTP request for RUM: https://github.com/axw/apm-server/blob/48b841ebe94a45c75d21f597d686ce7bbd3caf53/beater/api/mux.go#L319-L336

**Review by simitt — APPROVED** [review]

**Comment by felixbarny** [comment]
Can/should agents also stop collecting these fields?

**Comment by axw** [comment]
@felixbarny agents can and probably should stop collecting `request.socket.encrypted`, but they should continue collecting `request.socket.remote_address`. I think at some point we might want to replace that with something more general, though. It's currently tied to HTTP requests, and doesn't need to be.

**Pull request merged**